src/overlays/actors/ovl_En_Dekunuts/z_en_dekunuts.c

```diff
@@ -71,85 +121,634 @@ static DamageTable D_808BEF64 = {
+            func_808BD78C(this);
+        } else {
+            if (this->actor.params == 1) {
+                func_808BE680(this);
+            } else if ((this->actor.params == 0) && (this->actor.xzDistToPlayer < 120.0f)) {
+                func_808BDC9C(this);
+            } else {
+                func_808BD870(this);
+            }
+        }
+    }
+
+    if (phi_v1 && ((this->actor.xzDistToPlayer > 160.0f) || (this->actor.params != 0)) &&
+        (((this->actor.params == 0) && (fabsf(this->actor.yDistToPlayer) < 120.0f)) ||
+         ((this->actor.params == 2) && (this->actor.yDistToPlayer > -60.0f)) || (this->actor.params == 1)) &&
+        ((this->unk_190 == 0) || (this->actor.xzDistToPlayer < 480.0f))) {
+        this->skelAnime.animPlaybackSpeed = 1.0f;
+    }
+}
+
+void func_808BD78C(EnDekunuts* this) {
+    SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06002FA4);
+    this->unk_190 = 2;
+    this->actionFunc = func_808BD7D4;
+}
+
+void func_808BD7D4(EnDekunuts* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (func_801378B8(&this->skelAnime, 0.0f)) {
+        if (this->unk_190 != 0) {
+            this->unk_190--;
+        }
+    }
+    if ((this->unk_190 == 0) ||
+        ((this->actor.xzDistToPlayer < 120.0f) && Player_GetMask(globalCtx) != PLAYER_MASK_STONE_MASK)) {
+        func_808BDC9C(this);
+    }
+}
+
+void func_808BD870(EnDekunuts* this) {
+    SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_0600326C, -3.0f);
+    if (this->actionFunc == func_808BDA4C) {
+        this->unk_190 = 0x1002;
+    } else {
+        this->unk_190 = 1;
+    }
+    this->actionFunc = func_808BD8D8;
+}
+
+void func_808BD8D8(EnDekunuts* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (func_801378B8(&this->skelAnime, 0.0f)) {
+        if (this->unk_190 != 0) {
+            this->unk_190--;
+        }
+    }
+
+    if (!(this->unk_190 & 0x1000)) {
+        Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 2, 0xE38);
+    }
+
+    if (this->unk_190 == 0x1000) {
+        if ((this->actor.xzDistToPlayer > 480.0f) ||
+            ((this->actor.params == 0) && (this->actor.xzDistToPlayer < 120.0f)) ||
+            (Player_GetMask(globalCtx) == PLAYER_MASK_STONE_MASK)) {
+            func_808BDC9C(this);
+        } else {
+            func_808BDA08(this);
+        }
+    } else if (this->unk_190 == 0) {
+        if (Player_GetMask(globalCtx) == PLAYER_MASK_STONE_MASK) {
+            func_808BDC9C(this);
+        } else {
+            func_808BDA08(this);
+        }
+    }
+}
+
+void func_808BDA08(EnDekunuts* this) {
+    SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_06000168);
+    this->unk_190 = this->unk_194;
+    this->actionFunc = func_808BDA4C;
+}
+
+void func_808BDA4C(EnDekunuts* this, GlobalContext* globalCtx) {
+    Player* player;
+    Vec3f sp58;
+    s16 pitch;
+    f32 z;
+    f32 y;
+    f32 x;
```

Comment:
```suggestion
    Vec3f pos;
```
Can you turn this into a Vec3f and maintain a match?